src/pages/index.tsx

```diff
@@ -4,19 +4,108 @@ import { Layout } from "../components/Layout";
 import { useRouter } from "next/router";
 import dynamic from "next/dynamic";
 import { Posts } from "../components/Post";
+import { api } from "../utils/api";
+import { useEffect, useState } from "react";
+import { BlogPostCategory } from "../types/blogPost";
+import SearchPost from "../components/SearchPost";
+import { usePagination } from "../hooks/usePagination";
+import { useDebounce } from "../hooks/useDebounce";
+import { Pagination } from "../components/Pagination";
 
 const HomePage: NextPage = () => {
+  const [paginationPage, setPaginationPage] = useState(0);
+  const [searchActive, SetSearchActive] = useState(false);
+  const [searchCategory, setSearchCategory] = useState<BlogPostCategory[]>([]);
+  const [searchWord, setSearchWord] = useState<string>("");
+  const debounce = useDebounce(searchWord, 500);
   const router = useRouter();
+
+  useEffect(() => {
+    const params = new URLSearchParams(window.location.search);
+
+    const searchParam = params.get("search");
+    if (searchParam) {
+      setSearchWord(searchParam);
+    }
+
+    const categoryParam = params.get("category");
+    if (categoryParam !== null) {
+      const validCategories = Object.values(BlogPostCategory);
+      if (validCategories.includes(categoryParam as BlogPostCategory)) {
+        const categoryArray: BlogPostCategory[] = [
+          categoryParam as BlogPostCategory,
+        ];
+        setSearchCategory(categoryArray);
+      }
+    }
+  }, []);
+
+  const POSTS_PER_PAGE = 8;
+
+  const { data, isLoading, error } = api.blogPost.getBlogPosts.useQuery({
```

Comment:
I think destructing query makes code more difficult to understand. Having `blogPostsQuery.data` is much more readable than just `data`, which can mean anything